HomeWork/Cashier.js

```diff
@@ -0,0 +1,32 @@
+import { Person } from "./Person.js";
+
+export class Cashier extends Person {
+    
+    #totalSales;
+
+    constructor({ birthDate, name, totalSales }) {
+        super({ birthDate, name});
+        this.#totalSales = totalSales;
+    }
+    
+    // scan the products, calcultate the initial sum, make the customer to pay
+    scanAndCash(customer) {
+        let intialSum = 0;
+        customer.shoppingCart.forEach(product => {
+            intialSum += product.price;
+        });
+        customer.payTheAmountOfMoney(intialSum);
+        this.#totalSales += intialSum;
```

Comment:
I think that you should also remove all the products from the customer's shopping cart (via public methods, not its private `shoppingCart` property that you've leaked via the getter) if the `. payTheAmountOfMoney()` method didn't throw any error.